drivers/hmis/spec/requests/hmis/create_enrollment_spec.rb

```diff
@@ -0,0 +1,141 @@
+  let(:c1) { create :hmis_hud_client, data_source: ds1 }
+  let(:c2) { create :hmis_hud_client, data_source: ds1 }
+  let(:c3) { create :hmis_hud_client, data_source: ds1 }
+  let(:test_input) do
+    {
+      project_id: p1.id,
+      start_date: Date.today.strftime('%Y-%m-%d'),
+      household_members: [
+        {
+          id: c1.id,
+          relationship_to_ho_h: Types::HmisSchema::Enums::RelationshipToHoH.enum_member_for_value(1).first,
+        },
+        {
+          id: c2.id,
+          relationship_to_ho_h: Types::HmisSchema::Enums::RelationshipToHoH.enum_member_for_value(2).first,
+        },
+        {
+          id: c3.id,
+          relationship_to_ho_h: Types::HmisSchema::Enums::RelationshipToHoH.enum_member_for_value(3).first,
+        },
+      ],
+    }
+  end
+
+  describe 'enrollment creation tests' do
+    before(:each) do
+      user.add_viewable(ds1)
+      post hmis_user_session_path(hmis_user: { email: user.email, password: user.password })
+    end
+
+    let(:mutation) do
+      <<~GRAPHQL
+        mutation CreateEnrollment($input: CreateEnrollmentInput!) {
+          createEnrollment(input: $input) {
+            enrollments {
+              id
+              entryDate
+              inProgress
+              client {
+                id
+              }
+              project {
+                id
+              }
+            }
+            errors {
+              id
+              attribute
+              message
+              fullMessage
+              type
+              options
+              __typename
+            }
+          }
+        }
+      GRAPHQL
+    end
+
+    it 'should create an enrollment successfully' do
+      response, result = post_graphql(input: test_input) { mutation }
+
+      expect(response.status).to eq 200
+      enrollments = result.dig('data', 'createEnrollment', 'enrollments')
+      errors = result.dig('data', 'createEnrollment', 'errors')
+      expect(enrollments).to be_present
+      expect(enrollments.count).to eq(3)
+      expect(errors).to be_empty
+    end
+
+    describe 'In progress tests' do
+      it 'should set things to in progress if we tell it to' do
+        response, result = post_graphql(input: test_input.merge(in_progress: true)) { mutation }
+        expect(response.status).to eq 200
+        enrollments = result.dig('data', 'createEnrollment', 'enrollments')
+        errors = result.dig('data', 'createEnrollment', 'errors')
+        expect(enrollments).to be_present
+        expect(enrollments.count).to eq(3)
+        expect(enrollments.pluck('inProgress').uniq).to eq([true])
+        expect(enrollments.pluck('project')).to all(be_present)
+        expect(errors).to be_empty
+        expect(Hmis::Hud::Enrollment.count).to eq(3)
+        expect(Hmis::Hud::Enrollment.in_progress.count).to eq(3)
+        expect(Hmis::Hud::Enrollment.where(project_id: nil).count).to eq(3)
```

Comment:
Can you add assertions for `Wip` records being created with the correct values?

Reply:
Done